src/test/java/picard/analysis/TheoreticalSensitivityTest.java

```diff
@@ -207,22 +206,182 @@ public void testHetSensDistributions() throws Exception {
+        final Histogram<Integer> depthHistogram = histograms.get(0);
+        final Histogram<Integer> qualityHistogram = histograms.get(1);
+
+        final double[] qualityDistribution = TheoreticalSensitivity.normalizeHistogram(qualityHistogram);
+        final double[] depthDistribution = TheoreticalSensitivity.normalizeHistogram(depthHistogram);
+
+        final double resultFromTS = TheoreticalSensitivity.theoreticalSensitivity(depthHistogram, qualityHistogram, sampleSize, 3, 0.5);
+        final double resultFromTHS = TheoreticalSensitivity.hetSNPSensitivity(depthDistribution, qualityDistribution, sampleSize, 3);
+
+        Assert.assertEquals(resultFromTS, resultFromTHS, tolerance);
+    }
+
+    @DataProvider(name = "callingThresholdDataProvider")
+    public Object[][] callingThreshold() {
+        return new Object[][] {
+                // These values were tested with an independent implementation in R.
+                // Test a transition due to a change in the logOddsThreshold
+                {100, 10, 10*20, .1, 5.8, true},
+                {100, 10, 10*20, .1, 5.9, false},
+
+                // Test a transition due to change in average base quality from 20 to 21
+                {100, 10, 10*21, .1, 6.2, true},
+                {100, 10, 10*20, .1, 6.2, false},
+
+                // Test a transition due to change in total depth
+                {115, 10, 10*21, .1, 6.2, false},
+                {114, 10, 10*21, .1, 6.2, true}
+        };
+    }
+
+    @Test(dataProvider = "callingThresholdDataProvider")
+    public void testCallingThreshold(final int totalDepth, final int altDepth, final double sumOfAltQualities, final double alleleFraction, final double logOddsThreshold, final boolean expectedCall) {
+        Assert.assertEquals(TheoreticalSensitivity.isCalled(totalDepth, altDepth, sumOfAltQualities, alleleFraction, logOddsThreshold), expectedCall);
+    }
+
+    @DataProvider(name = "sumOfGaussiansDataProvider")
+    public Object[][] sumOfGaussians() {
+        final File wgsMetricsFile = new File(TEST_DIR, "test_Solexa-332667.wgs_metrics");
+        final File targetedMetricsFile = new File(TEST_DIR, "test_25103070136.targeted_pcr_metrics");
+
+        // When we sum more base qualities from a particular distribution, it should look increasingly Gaussian.
+        return new Object[][]{
+                {wgsMetricsFile, 500, 0.03},
+                {wgsMetricsFile, 20,  0.05},
+                {wgsMetricsFile, 10,  0.10},
+                {targetedMetricsFile, 500, 0.03},
+                {targetedMetricsFile, 20, 0.05},
+                {targetedMetricsFile, 10, 0.10}
+        };
+    }
+
+    @Test(dataProvider = "sumOfGaussiansDataProvider")
+    public void testDrawSumOfQScores(final File metricsFile, final int altDepth, final double tolerance) throws Exception {
+        final MetricsFile<TheoreticalSensitivityMetrics, Integer> Metrics = new MetricsFile<>();
+        Metrics.read(new FileReader(metricsFile));
```

Comment:
FileReaders should be opened using try-with-resources so that they are closed for sure.

Reply:
fixed